lib/ricohapi/mstorage/client.rb

```diff
@@ -40,8 +42,27 @@ def download(media_id)
       # GET /media/{id}/meta, GET /media/{id}/meta/exif, GET /media/{id}/meta/gpano,
       # GET /media/{id}/meta/user, GET /media/{id}/meta/user/{key}
       def meta(media_id, field_name = nil)
-        handle_response do
-          token.get endpoint_for("media/#{media_id}/meta")
+        unless field_name
+          # GET /media/{id}/meta
+          handle_response do
+            token.get endpoint_for("media/#{media_id}/meta")
+          end
+        else
+          if ['exif', 'gpano', 'user'].include? field_name
+            # GET /media/{id}/meta/exif, GET /media/{id}/meta/gpano, GET /media/{id}/meta/user
+            handle_response do
+              token.get endpoint_for("media/#{media_id}/meta/#{field_name}")
+            end
+          else
+            if User_meta_regex =~ field_name
+              # GET /media/{id}/meta/user/{key}
+              handle_response(:as_raw) do
+                token.get endpoint_for("media/#{media_id}/meta/user/#{$1}")
+              end
+            else
+              raise Error.new("invalid fieldName: #{field_name}")
+            end
+          end
```

Comment:
I didn't know this way, and I think it's much simpler!
I confirmed to get matched substring by `$1`, so I'll fix as you explained.